applications/StructuralMechanicsApplication/custom_elements/beam_elements/timoshenko_beam_element_3D2N.cpp

```diff
@@ -0,0 +1,932 @@
+        rElementalDofList[index]     = r_geom[i].pGetDof(DISPLACEMENT_X);
+        rElementalDofList[index + 1] = r_geom[i].pGetDof(DISPLACEMENT_Y);
+        rElementalDofList[index + 2] = r_geom[i].pGetDof(DISPLACEMENT_Z);
+        rElementalDofList[index + 3] = r_geom[i].pGetDof(ROTATION_X    );
+        rElementalDofList[index + 4] = r_geom[i].pGetDof(ROTATION_Y    );
+        rElementalDofList[index + 5] = r_geom[i].pGetDof(ROTATION_Z    );
+    }
+
+    KRATOS_CATCH("")
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+array_1d<double, 3> LinearTimoshenkoBeamElement3D2N::GetLocalAxesBodyForce(
+    const Element &rElement,
+    const GeometryType::IntegrationPointsArrayType &rIntegrationPoints,
+    const IndexType PointNumber
+    ) const
+{
+    const auto body_force = StructuralMechanicsElementUtilities::GetBodyForce(*this, rIntegrationPoints, PointNumber);
+    BoundedMatrix<double, 3, 3> T;
+    noalias(T) = StructuralMechanicsElementUtilities::GetFrenetSerretMatrix3D(GetGeometry());
+    array_1d<double, 3> local_body_force = prod(T, body_force);
+    return local_body_force;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+BoundedMatrix<double, 3, 3> LinearTimoshenkoBeamElement3D2N::GetConsistentFrenetSerretMatrix3D(
+    const GeometryType& rGeometry
+    ) const
+{
+    BoundedMatrix<double, 3, 3> T;
+    noalias(T) = StructuralMechanicsElementUtilities::GetFrenetSerretMatrix3D(rGeometry);
+    return T;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+
+void LinearTimoshenkoBeamElement3D2N::GetNodalValuesVector(
+    VectorType& rNodalValues
+    ) const
+{
+    KRATOS_TRY
+
+    const auto& r_geom = GetGeometry();
+    const SizeType num_nodes = r_geom.size();
+    const SizeType global_size = GetDoFsPerNode() * num_nodes;
+
+    if (rNodalValues.size() != global_size)
+        rNodalValues.resize(global_size, false);
+
+    BoundedMatrix<double, 3, 3> T;
+    noalias(T) = GetConsistentFrenetSerretMatrix3D(r_geom);
+
+    const auto& r_displ_0    = r_geom[0].FastGetSolutionStepValue(DISPLACEMENT);
+    const auto& r_rotation_0 = r_geom[0].FastGetSolutionStepValue(ROTATION);
+    const auto& r_displ_1    = r_geom[1].FastGetSolutionStepValue(DISPLACEMENT);
+    const auto& r_rotation_1 = r_geom[1].FastGetSolutionStepValue(ROTATION);
+
+    // Here we rotate the vectors to local axes
+    const VectorType& r_local_displ_0 = prod(T, r_displ_0);
+    const VectorType& r_local_displ_1 = prod(T, r_displ_1);
+
+    // Due to dextrogyr axes system
+    T(0, 1) *= -1.0;
+    T(1, 1) *= -1.0;
+    T(2, 1) *= -1.0;
```

Comment:
I would like to avoid that a modified T is accidentally used for displacements This can be expressed by making a copy and then modifying:
```suggestion
    // Due to dextrogyr axes system
    auto T_rotation = T;
    column(T_rotation, 1) *= -1.0;
```
The column function expresses the intention more clearly than component by component.

Am I correct when I assume that the definition of the Y axis rotation D.o.F. in the nodes cause this? Can you point me to the code where the rotation D.o.F. of the overall system are defined, I 'm searching but without success so far.